web/html/src/manager/minion/ansible/ansible-control-node.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import SpaRenderer from "core/spa/spa-renderer";
+import { Messages, Utils } from "components/messages";
+import { TextField } from "components/fields";
+import { Panel } from "components/panels/Panel";
+import { AsyncButton, Button } from "components/buttons";
+import Network from "utils/network";
+import NewAnsiblePath from "./new-ansible-path";
+import EditAnsiblePath from "./edit-ansible-path";
+
+type AnsiblePath = {
+  id?: number;
+  minionServerId?: number;
+  type?: string;
+  path?: string;
+}
+
+type PropsType = {
+  minionServerId: number;
+};
+
+type StateType = {
+  minionServerId: number;
+  playbooksPaths: AnsiblePath[];
+  inventoriesPaths: AnsiblePath[];
+  newPlaybookPath: string;
+  newInventoryPath: string;
+  editPlaybookPath?: AnsiblePath;
+  editInventoryPath?: AnsiblePath;
+  errors: string[];
+};
+
+class AnsibleControlNode extends React.Component<PropsType, StateType> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      minionServerId: props.minionServerId,
+      playbooksPaths: [],
+      inventoriesPaths: [],
+      newPlaybookPath: "",
+      newInventoryPath: "",
+      editPlaybookPath: undefined,
+      editInventoryPath: undefined,
+      errors: [],
+    };
+
+    Network.get("/rhn/manager/api/systems/details/ansible/paths/" + props.minionServerId)
+    .promise.then(data => {
+      this.setState({ playbooksPaths: data.filter(p => p.type === "playbook"), inventoriesPaths: data.filter(p => p.type === "inventory") });
+    });
+  }
+
+  newPath(type: string, newPath: string) {
+    if (type === "playbook") {
+      this.setState({ newPlaybookPath: newPath });
+    }
+    else {
+      this.setState({ newInventoryPath: newPath });
+    }
+  }
+
+  deletePath(path: AnsiblePath) {
+    Network.post(
+      "/rhn/manager/api/systems/details/ansible/paths/delete",
+      JSON.stringify(path.id),
+      "application/json"
+    ).promise.then(data => {
+      if (data.success) {
+        if (path.type === "playbook") {
+          this.setState({ playbooksPaths: this.state.playbooksPaths.filter(p => p.id !== path.id) });
```

Comment:
I'm not sure what the business logic is here, so checking just in case: is it intentional that `inventoriesPaths` is not set to an empty array in this branch and vice versa in the other one?

Reply:
Yes it is intentional: we are deleting only one entry which can be only in one or in the other set of paths, indeed we enter in an if condition. I could think about something more abstract where we filter both sets by dropping the path we deleted, but in the end it will be just a drop in one of the two, and the other one will not be affected at all. The logic maybe is too narrowed down when I set the state at the beginning by filtering the whole set into two subset sliced by the type. Anyway, this was intentional. Thanks for the heads-up :+1: